test/e2e/storage/dax/dax.go

```diff
@@ -359,3 +361,127 @@ func testDaxOutside(
+		Framework: f,
+	}
+	fstype := ""
+	vsource := v1.VolumeSource{
+		CSI: &v1.CSIVolumeSource{
+			Driver: "pmem-csi.intel.com",
+			FSType: &fstype,
+			VolumeAttributes: map[string]string{
+				"size": "110Mi",
+			},
+		},
+	}
+	It("should cause hugepage paging event with default fs", func() {
+		init()
+		testHugepageInPod(f, l.root, &vsource, config)
+	})
+	/* there is issue in kubelet causing panic and retry loop when fsType is set to ext4 or xfs.
+		   The following 2 items can be enabled after that gets fixed.
+	           https://github.com/kubernetes/kubernetes/issues/102651
+		        It("should cause hugepage paging event with ext4", func() {
+				init()
+				fsExt4 := "ext4"
+				vsource.CSI.FSType = &fsExt4
+				testHugepageInPod(f, l.root, &vsource, config)
+			})
+			It("should cause hugepage paging event with xfs", func() {
+				init()
+				fsXFS := "xfs"
+				vsource.CSI.FSType = &fsXFS
+				testHugepageInPod(f, l.root, &vsource, config)
+			})*/
+})
+
+func testHugepageInPod(
+	f *framework.Framework,
+	root string,
+	source *v1.VolumeSource,
+	config *testsuites.PerTestConfig,
+) {
+	pod := CreatePod(f, "hugepage-test", v1.PersistentVolumeFilesystem, source, config, false)
+	defer func() {
+		DeletePod(f, pod)
+	}()
+	testHugepage(f, pod, root, source)
+	DeletePod(f, pod)
+}
+
+func testHugepage(
+	f *framework.Framework,
+	pod *v1.Pod,
+	root string,
+	source *v1.VolumeSource,
+) {
+	ns := f.Namespace.Name
+	containerName := pod.Spec.Containers[0].Name
+
+	// run trace monitor on all workers
+	for worker := 1; ; worker++ {
+		sshcmd := fmt.Sprintf("%s/_work/%s/ssh.%d", os.Getenv("REPO_ROOT"), os.Getenv("CLUSTER"), worker)
+		ssh := exec.Command(sshcmd, "sudo sh -c 'echo 1 > /sys/kernel/debug/tracing/events/fs_dax/dax_pmd_fault_done/enable; echo 1 > /sys/kernel/debug/tracing/tracing_on; cat /sys/kernel/debug/tracing/trace_pipe > /tmp/tracetmp 2>&1 &'")
+		_, err := ssh.Output()
+		if err != nil && os.IsNotExist(err) {
+			break
+		}
+	}
+
+	accessOutput, _ := pmempod.RunInPod(f, root, []string{accessHugepagesBinary}, accessHugepagesBinary, ns, pod.Name, containerName)
+	By(fmt.Sprintf("Output from pmem-access-hugepages pod:[%s]", accessOutput))
+	for worker := 1; ; worker++ {
+		sshcmd := fmt.Sprintf("%s/_work/%s/ssh.%d", os.Getenv("REPO_ROOT"), os.Getenv("CLUSTER"), worker)
+		ssh := exec.Command(sshcmd, "sudo sh -c 'echo 0 > /sys/kernel/debug/tracing/events/fs_dax/dax_pmd_fault_done/enable; echo 0 > /sys/kernel/debug/tracing/tracing_on; pkill -f cat\\ /sys/kernel/debug/tracing/trace_pipe'")
+		_, err := ssh.Output()
+
+		// There may be garbage (zero values) at start, we get better results by counting from end, thats why we use NF-relative fields in awk
+		ssh = exec.Command(sshcmd, "cat /tmp/tracetmp|awk '{print $(NF-13) $(NF-9)}'")
+		traceOutput, err := ssh.Output()
+		if err != nil && os.IsNotExist(err) {
+			// if we are here, no worker node had tracer output
+			By("Unexpected: no worker has tracer output?")
```

Comment:
Shouldn't that be a test failure?

No-one is going to read the CI output to find this message unless the test is marked as failed...

Reply:
turned that into failure